internal/pkg/runtime/engines/singularity/container_linux.go

```diff
@@ -1742,3 +1713,56 @@ func (c *container) addActionsMount(system *mount.System) error {
 
 	return system.Points.AddRemount(mount.BindsTag, containerDir, flags)
 }
+
+func (c *container) prepareNetworkSetup(system *mount.System, pid int) (func() error, error) {
+	if c.netNS {
+		if os.Geteuid() == 0 && !c.userNS {
+			// we hold a reference to container network namespace
+			// by binding /proc/self/ns/net (from RPC server) inside
+			// session directory
+			if err := c.session.AddFile("/netns", nil); err != nil {
+				return nil, err
+			}
+			nspath, _ := c.session.GetPath("/netns")
+			if err := system.Points.AddBind(mount.SharedTag, "/proc/self/ns/net", nspath, 0); err != nil {
+				return nil, fmt.Errorf("could not hold network namespace reference: %s", err)
+			}
+			networks := strings.Split(c.engine.EngineConfig.GetNetwork(), ",")
+
+			cniPath := &network.CNIPath{}
+
+			if c.engine.EngineConfig.File.CniConfPath != "" {
+				cniPath.Conf = c.engine.EngineConfig.File.CniConfPath
+			} else {
+				cniPath.Conf = defaultCNIConfPath
+			}
+			if c.engine.EngineConfig.File.CniPluginPath != "" {
+				cniPath.Plugin = c.engine.EngineConfig.File.CniPluginPath
+			} else {
+				cniPath.Plugin = defaultCNIPluginPath
```

Comment:
same as above, that will simplify code a bit, in my opinion
in general `else` is not used much in go, so getting rid of it will make code more 'native'